app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb

```diff
@@ -133,7 +133,7 @@ def perf_capture_data_amazon(start_time, end_time)
     COUNTER_INFO.each do |i|
       timestamps = i[:amazon_counters].collect do |c|
         metrics_by_counter_name[c].keys unless metrics_by_counter_name[c].nil?
-      end.flatten.uniq.sort.compact
+      end.flatten.uniq.compact.sort
```

Comment:
@Fryguy this new example tests this

https://github.com/ManageIQ/manageiq/pull/4718/files#diff-ffbdff4705cd8ab74bed2c08ee3d8d9cR32